simulation.cpp

```diff
@@ -673,6 +673,47 @@ void test_simulation() noexcept//!OCLINT test may be many
     }
 #endif
 
+    ///#263
+    ///Selection can happen sporadically every n generations
+    {
+        int selection_freq = 5;
+        int repeats = 10;
+
+        sim_param s_p{};
+        s_p.n_generations = repeats;
+        s_p.selection_freq = selection_freq;
+        pop_param p_p;
+        p_p.number_of_inds = 1000;
+        p_p.mut_rate_weight = 0.5;
+        p_p.mut_step = 0.1;
+        all_params a_p{{},{}, p_p, s_p};
+
+        simulation<population<>,
+                env_change_type::symmetrical,
+                selection_type::sporadic> s{a_p};
+
+        double stdev_prev_pop;
+        double stdev_pop;
+        double avg_pop;
+        double avg_prev_pop;
+
+        for (int i = 0; i != repeats; i++)
+        {
+            tick(s);
+
+            if(s.get_time() % s.get_sel_freq() == 0  && s.get_time() != 0)
```

Comment:
Here you check that this happens when you want it to happen, but If i'm not mistaken that test would already have passed when selection happens at every generation. I think you also need to check selection does *not* happen when you don't want it to

Reply:
true